### ISSUE TITLE
Psalm silently ignores errors

### ISSUE THREAD

**Issue by rarila** [vimeo/psalm#11266]
While working on a Symfony project I suddenly was confronted with ``@psalm-trace`` outputting nothing on certain files. Then I noticed that Psalm even did not report a single error except of parse errors. After further examination I found out, that this happens with Repository files.

To reproduce from clean start do the following (I assume this happens with every Symfony projects, if you have one up you can simply try there):

> ``composer create-project symfony/skeleton:"6.4.x" test``
> ``cd test``
> ``composer require webapp vimeo/psalm``
> ``.\vendor\bin\psalm --init``
> set ``findUnusedCode`` to ``false`` in psalm.xml

Add a file ``TestRepository.php`` with intentional errors in the sourcefolder:

```php
<?php

namespace App;

use Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepository;

final class TestRepository extends ServiceEntityRepository
{
    public function __construct()
    {
        $test = '';
        /** @psalm-trace $test */;

        thisFails();
    }
}
```

Finally run  ``.\vendor\bin\psalm`` and you end up with

> No errors found!

instead of expected ``Trace`` and ``UndefinedFunction``.

So the problem is with extending [ServiceEntityRepository](https://github.com/doctrine/DoctrineBundle/blob/2.13.x/src/Repository/ServiceEntityRepository.php). If you comment out one of the class definitions in that file, Psalm executes as expected and reports the errors.

The problem happens with latest v5 Psalm to latest v6 Psalm. I guess this applies not only to ServiceEntityRepository of DoctrineBundle, but also to other dependencies using such a construct.

**Comment by zmitic** [comment]
As a temporary solution, you can extend `ServiceEntityRepositoryProxy` which is the same thing as what [ORM 3 does](https://github.com/doctrine/DoctrineBundle/blob/2.13.x/src/Repository/ServiceEntityRepository.php#L50). Psalm would then complain about using internal methods so you can do this in issue handlers:

```xml
<InternalClass>
    <errorLevel type="suppress">
        <referencedClass name="Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepositoryProxy"/>
    </errorLevel>
</InternalClass>
<InternalMethod>
    <errorLevel type="suppress">
        <referencedMethod name="Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepositoryProxy::find"/>
        <referencedMethod name="Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepositoryProxy::findAll"/>
        <referencedMethod name="Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepositoryProxy::findBy"/>
        <referencedMethod name="Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepositoryProxy::findOneBy"/>
        <referencedMethod name="Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepositoryProxy::count"/>
        <referencedMethod name="Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepositoryProxy::createQueryBuilder"/>
        <referencedMethod name="Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepositoryProxy::__construct"/>
        <referencedMethod name="Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepositoryProxy::getEntityManager"/>
    </errorLevel>
</InternalMethod>
```

It ain't pretty, it is a copy&paste of my current setup so there might be something missing. But this is probably the only way until the support for Doctrine 2 is dropped. Based on[ this comment](https://github.com/doctrine/DoctrineBundle/pull/1756#discussion_r1490574964), it seems that PHPStorm, PHPStan and csfixer all have the same problem.

**Comment by rarila** [comment]
@zmitic That's exactly what I did. But anyway, thanks a lot for your efforts. 👍 

But at least PHPstan does report the other errors (If I remember correctly – I didn’t check if it got the import right though). So the problem is, if you only use Psalm and are not aware of this problem you can have lot's of errors in your repository code and think everything is ok. 💥

**Comment by rarila** [comment]
Just checked… PHPStan reports:

> • Class App\Test extends generic class Doctrine\...\ServiceEntityRepository but does not specify its types: T
> • Function thisFails not found.

Further investigation lets me assume, PHPStan just take the first class definition (which is wrong in my test case, as ORM 3 is installed) and ignores the second one.
But at least the other errors are reported.

**Comment by ro0NL** [comment]
ref https://github.com/doctrine/DoctrineBundle/issues/1677

ideally psalm fixes this

**Comment by ro0NL** [comment]
an ORM v3 fix is to extend `ServiceEntityRepositoryProxy` instead

**Comment by ro0NL** [comment]
cc @danog also, psalm is missing all doctrine bundle repositories for quite some time now, and it hurts bumping orm
